db/versions/0015.yml

```
@@ -0,0 +1,186 @@
+            'updated', updated,
+            'installationId', installation_id,
+            'eventType', event_type,
+            'eventId', event_id) as value,
+          1 as version,
+          github_builds.etag as etag
+      from github_builds
+      where
+        github_builds.task_group_id = decode_string_key(taskcluster_github_builds_entities_load.partition_key);
+      end
+  taskcluster_github_builds_entities_create:
+    description: See taskcluster-lib-entities
+    serviceName: github
+    mode: write
+    args: pk text, rk text, properties jsonb, overwrite boolean, version integer
+    returns: uuid
+    body: |-
+      declare
+        new_row github_builds%ROWTYPE;
+      begin
+        select
+          (properties ->> 'organization')::text as organization,
+          (properties ->> 'repository')::text as repository,
+          (properties ->> 'sha')::text as sha,
+          (properties ->> 'taskGroupId')::text as task_group_id,
+          (properties ->> 'state')::text as state,
+          (properties ->> 'created')::timestamptz as created,
+          (properties ->> 'updated')::timestamptz as updated,
+          (properties ->> 'installationId')::integer as installation_id,
+          (properties ->> 'eventType')::text as event_type,
+          (properties ->> 'eventId')::text as event_id,
+          public.gen_random_uuid() as etag
+        into new_row;
+        if overwrite then
+          raise exception 'overwrite not implemented';
+        else
+          execute 'insert into github_builds select $1.*' using new_row;
+        end if;
+        return new_row.etag;
+      end
+  taskcluster_github_builds_entities_scan:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: github
+    args: pk text, rk text, condition text, size integer, page integer
+    returns: table (partition_key text, row_key text, value jsonb, version integer, etag uuid)
+    body: |-
+      declare
+      begin
+        if not condition is null then
+          raise exception 'condition not supported';
+        end if;
+        return query
+          select
+            taskcluster_github_builds_entities_scan.partition_key,
+            'taskGroupId' as row_key,
+            jsonb_build_object(
+              'PartitionKey', encode_string_key(task_group_id),
+              'RowKey', 'taskGroupId',
+              'organization', organization,
+              'repository', repository,
+              'sha', sha,
+              'taskGroupId', task_group_id,
+              'state', state,
+              'created', created,
+              'updated', updated,
+              'installationId', installation_id,
+              'eventType', event_type,
+              'eventId', event_id) as value,
+              1 as version,
+          github_builds.etag as etag
+          from github_builds
+          where
+            partition_key is null or
+            task_group_id = decode_string_key(task_group_id)
```

```suggestion
            task_group_id = decode_string_key(partition_key)
```
It looks like no calls to `.scan` or `.query` pass a partition key, so it's not too surprising this didn't cause a test failure, and wouldn't cause a failure in production either.  You could verify by adding a check that partition_key is null in this function.

---

Made this change in my branch